src/Persistence/Proxy/PersistedObjectsTracker.php

```diff
@@ -63,8 +67,9 @@ static function(\WeakReference $weakRef) {
                     }
 
                     $clone = clone $object;
-                    $reflector->resetAsLazyGhost($object, function($object) use ($clone) {
-                        Configuration::instance()->persistence()->autorefresh($object, $clone);
+                    $id = Configuration::instance(checkFactoriesTrait: false)->persistence()->getIdentifierValues($object);
```

Comment:
We need to call `Configuration::instance()->persistence()->getIdentifierValues($object)` in `PersistedObjectsTracker::proxifyObjects()`. But this was messing with the container, during the `kernel.reset` phase. The problem is more or less the same than explained in [this issue](https://github.com/zenstruck/foundry/issues/963) - BTW, I've fixed this issue, and created the test mentioned there. 

Hence the new parameter `bool $checkFactoriesTrait` in `Configuratoin::instance()`

any way, this check will be removed in Foundry 3